AppBuilder/platform/process/tasks/ABProcessTaskSubProcess.js

```diff
@@ -0,0 +1,178 @@
+      let ids = this.propertyIDs(id);
+
+      // Pull query tasks option list
+      let parameterOptions = (this.process.processDataFields(this) || []).map(
+         (item) => {
+            return {
+               id: item.key,
+               value: item.label,
+            };
+         }
+      );
+
+      let ui = {
+         id: id,
+         view: "form",
+         elementsConfig: {
+            labelWidth: 120,
+         },
+         elements: [
+            {
+               id: ids.name,
+               view: "text",
+               label: L("Name"),
+               name: "name",
+               value: this.name,
+            },
+            {
+               id: ids.isEnable,
+               view: "switch",
+               label: L("Enable"),
+               value: this.isEnable,
+            },
+            {
+               id: ids.parameterId,
+               view: "richselect",
+               label: L("Repeat for"),
+               options: parameterOptions,
+               value: this.parameterId,
+            },
+         ],
+      };
+
+      webix.ui(ui, $$(id));
+
+      $$(id).show();
+   }
+
+   /**
+    * @method propertiesStash()
+    * pull our values from our property panel.
+    * @param {string} id
+    *        the webix $$(id) of the properties panel area.
+    */
+   propertiesStash(id) {
+      let ids = this.propertyIDs(id);
+      this.name = this.property(ids.name);
+      this.isEnable = this.property(ids.isEnable);
+      this.parameterId = this.property(ids.parameterId);
+   }
+
+   /**
+    * @method diagramProperties()
+    * return a set of values for the XML shape definition based upon
+    * the current values of this object.
+    * @return {json}
+    */
+   diagramProperties(bpmnModeler) {
+      let props = super.diagramProperties();
+      props = props || [{}];
+      props[0].def = props[0].def || {};
+
+      if (this.parameterId != null) {
+         props[0].def = props[0].def || {};
+         props[0].def.loopCharacteristics = bpmnModeler
+            .get("moddle")
+            .create("bpmn:MultiInstanceLoopCharacteristics");
+         props[0].def.loopCharacteristics.isSequential = true;
+      }
+
+      return props;
+   }
+
+   /**
+    * @method onChange()
+    * update the current Task with information that was relevant
+    * from the provided BPMN:Element
+    * @param {BPMNElement}
+    */
+   onChange(defElement) {
+      let loopCharacteristics =
+         defElement.loopCharacteristics ||
+         defElement.businessObject.loopCharacteristics ||
+         {};
+
+      switch (loopCharacteristics.$type) {
+         case "bpmn:MultiInstanceLoopCharacteristics":
+            this.loopType = loopCharacteristics.isSequential
+               ? "sequential"
+               : "parallel";
+            break;
+         case "bpmn:StandardLoopCharacteristics":
+            this.loopType = "looping";
+            break;
+      }
+   }
+
+   /**
+    * @method destroy()
```

Comment:
ok, these comments don't go along with these methods.  Need to write better comments here.

Explain why we are calling the parent `.process.xxx()` functions.